packages/@atjson/document/src/annotations/slice.ts

```diff
@@ -11,6 +11,19 @@ import { Annotation } from "../internals";
  */
 export class SliceAnnotation extends Annotation<{
   refs: string[];
+  /**
+   * Whether the range of the document should
+   * be retained in rendering. If true, the
+   * slice can be referred to and it will display
+   * in situ in the document.
+   *
+   * This is useful for cases like progressive
+   * enhancement of documents, where navigation
+   * is created dynamically from items in the document,
+   * and elements from the page are used for navigation
+   * purposes.
+   */
+  retain?: boolean;
```

Comment:
Where/when would we set this? Shouldn't there be some corresponding converter where I see this?

Reply:
Oh, this is a new feature intended for slices to be kept in place instead of extracted from the document.

This would be useful for features where you'd like to have some navigation that's extracted from text in the document (for example, headlines that are used to separate sections).